numbagg/test/test_grouped.py

```diff
@@ -3,10 +3,124 @@
 import pytest
 from numpy.testing import assert_almost_equal
 
-from numbagg.grouped import group_nanmean, group_nansum
+from numbagg.grouped import (
+    group_nanargmax,
+    group_nanargmin,
+    group_nancount,
+    group_nanfirst,
+    group_nanlast,
+    group_nanmean,
+    group_nanprod,
+    group_nansum,
+    group_nansum_of_squares,
+)
 
-ALL_FUNCS = [group_nanmean, group_nansum]
-NP_FUNCS = [np.mean, np.sum]
+FUNCTIONS = [
+    (group_nanmean, lambda x: x.mean(), np.nanmean),
+    (group_nansum, lambda x: x.sum(), np.nansum),
+    (group_nancount, lambda x: x.count(), None),
+    (group_nanfirst, lambda x: x.first(), None),
+    (group_nanlast, lambda x: x.last(), None),
+    (group_nanprod, lambda x: x.prod(), np.nanprod),
+    (group_nanargmax, lambda x: x.idxmax(), np.nanargmax),
+    (group_nanargmin, lambda x: x.idxmin(), np.nanargmin),
+    (
+        group_nansum_of_squares,
+        lambda x: x.agg(lambda y: (y**2).sum()),
+        lambda x: np.nansum(x**2),
+    ),
+]
+
+# Functions which return the same scalar if one is passed
+FUNCTIONS_CONSTANT = [
+    (group_nanmean, lambda x: x.mean(), np.nanmean),
+    (group_nansum, lambda x: x.sum(), np.nansum),
+    (group_nanfirst, lambda x: x.first(), None),
+    (group_nanlast, lambda x: x.last(), None),
+    (group_nanprod, lambda x: x.prod(), np.nanprod),
+]
+
+
+@pytest.fixture(scope="module")
+def rs():
+    return np.random.RandomState(0)
+
+
+@pytest.fixture(params=[np.float64, np.int32], scope="module")
+def dtype(request):
+    return request.param
+
+
+@pytest.fixture(scope="module")
+def values(rs, dtype):
+    if dtype == np.int32:
+        return rs.randint(0, 100, size=200)
+    elif dtype == np.float64:
+        vals = rs.rand(200)
+        return np.where(vals > 0.1, vals, np.nan)
+    else:
+        raise ValueError(f"dtype {dtype} not supported")
+
+
+@pytest.fixture()
+def labels(rs):
+    labels = rs.choice([0, 1, 2, 3, 4, 5], size=200)
+    # The tests are dependent on this being dense
+    assert len(np.unique(labels)) == 6
+    return labels
+
+
+@pytest.mark.parametrize("numbagg_func, pandas_func, _", FUNCTIONS)
+def test_group_pandas_comparison(values, labels, numbagg_func, pandas_func, _, dtype):
+    expected = pandas_func(pd.Series(values).groupby(labels))
+    result = numbagg_func(values, labels)
+    if dtype == np.int32:
+        if numbagg_func == group_nanprod:
+            pytest.skip("group_nanprod result too large")
+        assert_almost_equal(result, expected.values.astype(np.int32))
```

Comment:
@dcherian if we pass in ints, we get ints back. 

I think that makes sense, since folks can always convert to floats. Is that consistent with your thinking?

(It's inconsistent with bools — we don't get bools back if we pass in bools...)